data/src/main/java/org/springframework/security/data/repository/query/SecurityEvaluationContextExtension.java

```diff
@@ -77,12 +84,21 @@
  * it.
  *
  * @author Rob Winch
+ * @author Evgeniy Cheban
  * @since 4.0
  */
 public class SecurityEvaluationContextExtension implements EvaluationContextExtension {
 
 	private Authentication authentication;
 
+	private AuthenticationTrustResolver trustResolver = new AuthenticationTrustResolverImpl();
+
+	private RoleHierarchy roleHierarchy = new NullRoleHierarchy();
+
+	private PermissionEvaluator permissionEvaluator = new DenyAllPermissionEvaluator();
+
+	private String defaultRolePrefix = "ROLE_";
```

Comment:
This also seems fine, and I noticed this pattern elsewhere in the codebase. However, since the `SecurityContextRoot` defaults to `"ROLE_"`, this seems duplicated. Could/should we reduce the number of places we specify the default role prefix by defaulting to `null` and checking whether to set it on the `SecurityContextRoot` if not `null`?

Reply:
Remains without changes.